### PR TITLE
api updates

### PR DESCRIPTION
**Pull request by brettsam**

### CODE REVIEW

File: samples/FunctionApp/Program.cs [hunk 1]
```diff
@@ -19,14 +16,8 @@ static async Task Main(string[] args)
...
             var host = new HostBuilder()
-                //<docsnippet_configure_app>
-                .ConfigureAppConfiguration(c =>
-                {
-                    c.AddCommandLine(args);
-                })
-                //</docsnippet_configure_app>
                 //<docsnippet_middleware>
-                .ConfigureFunctionsWorkerDefaults()
+                .ConfigureFunctionsWorkerDefaults(args)
```
Comment (brettsam) [inline samples/FunctionApp/Program.cs:20]: @fabiocav -- wanted to make sure we were all cool with this before i update doc comments...

Comment (fabiocav) [reply samples/FunctionApp/Program.cs:20]: Had a discussion offline

**Review by fabiocav — APPROVED** [review]
Approved, pending the stuff that we discussed.

File: src/DotNetWorker/Hosting/WorkerHostBuilderExtensions.cs [hunk 2]
```diff
@@ -108,16 +109,28 @@ public static IHostBuilder ConfigureFunctionsWorkerDefaults(this IHostBuilder bu
...
+                .ConfigureAppConfiguration(configBuilder =>
+                {
+                    // If it does not begin with a key prefix of '--', wrap the first argument
+                    // in quotes so that nothing else is interpreted as a key prefix.
+                    var cmdLine = Environment.GetCommandLineArgs();
+                    if (cmdLine.Length > 0 && !cmdLine[0].StartsWith("--"))
+                    {
+                        cmdLine[0] = $"\"{cmdLine[0]}\"";
+                    }
+                    configBuilder.AddCommandLine(cmdLine);
```
Comment (brettsam) [inline src/DotNetWorker/Hosting/WorkerHostBuilderExtensions.cs:122]: Here's the magic.

**Review by fabiocav — APPROVED** [review]

**Pull request merged**